sockets_extension.go

```diff
@@ -412,3 +323,71 @@ func (IntValue) sockopt() {}
 func (i IntValue) String() string {
 	return strconv.Itoa(int(i))
 }
+
+// SocketsNotSupported is a helper type intended to be embeded in
+// implementations of the Sytem interface that do not support sockets.
+//
+// The type defines all socket-related methods to return ENOSYS, allowing
+// the type to implement the interface but indicating to callers that the
+// functionality is not supported.
+type SocketsNotSupported struct{}
+
+func (SocketsNotSupported) SockOpen(ctx context.Context, family ProtocolFamily, socketType SocketType, protocol Protocol, rightsBase, rightsInheriting Rights) (FD, Errno) {
+	return -1, ENOSYS
+}
+
+func (SocketsNotSupported) SockBind(ctx context.Context, fd FD, addr SocketAddress) (SocketAddress, Errno) {
+	return nil, ENOSYS
+}
+
+func (SocketsNotSupported) SockConnect(ctx context.Context, fd FD, addr SocketAddress) (SocketAddress, Errno) {
+	return nil, ENOSYS
+}
+
+func (SocketsNotSupported) SockListen(ctx context.Context, fd FD, backlog int) Errno {
+	return ENOSYS
+}
+
+func (SocketsNotSupported) SockAccept(ctx context.Context, fd FD, flags FDFlags) (FD, SocketAddress, SocketAddress, Errno) {
+	return -1, nil, nil, ENOSYS
+}
+
+func (SocketsNotSupported) SockRecv(ctx context.Context, fd FD, iovecs []IOVec, flags RIFlags) (Size, ROFlags, Errno) {
+	return 0, 0, ENOSYS
+}
+
+func (SocketsNotSupported) SockSend(ctx context.Context, fd FD, iovecs []IOVec, flags SIFlags) (Size, Errno) {
+	return 0, ENOSYS
+}
+
+func (SocketsNotSupported) SockSendTo(ctx context.Context, fd FD, iovecs []IOVec, flags SIFlags, addr SocketAddress) (Size, Errno) {
+	return 0, ENOSYS
+}
+
+func (SocketsNotSupported) SockRecvFrom(ctx context.Context, fd FD, iovecs []IOVec, flags RIFlags) (Size, ROFlags, SocketAddress, Errno) {
+	return 0, 0, nil, ENOSYS
+}
+
+func (SocketsNotSupported) SockGetOpt(ctx context.Context, fd FD, level SocketOptionLevel, option SocketOption) (SocketOptionValue, Errno) {
+	return nil, ENOSYS
+}
+
+func (SocketsNotSupported) SockSetOpt(ctx context.Context, fd FD, level SocketOptionLevel, option SocketOption, value SocketOptionValue) Errno {
+	return ENOSYS
+}
+
+func (SocketsNotSupported) SockLocalAddress(ctx context.Context, fd FD) (SocketAddress, Errno) {
+	return nil, ENOSYS
+}
+
+func (SocketsNotSupported) SockRemoteAddress(ctx context.Context, fd FD) (SocketAddress, Errno) {
+	return nil, ENOSYS
+}
+
+func (SocketsNotSupported) SockAddressInfo(ctx context.Context, name, service string, hints AddressInfo, results []AddressInfo) (int, Errno) {
+	return 0, ENOSYS
+}
+
+func (SocketsNotSupported) SockShutdown(ctx context.Context, fd FD, flags SDFlags) Errno {
+	return ENOSYS
+}
```

Comment:
Let's combine `socket.go` and `sockets_extension.go` into one file? (e.g. `sockets.go`)?

Reply:
I kept the file name `socket.go` (singular) for consistency with all the other source files, none are plural names.